docs/docsgen/source/technical/float8.rst

```diff
@@ -0,0 +1,168 @@
+
+Two papers have been published in 2022 to introduce floats
+stored on a byte as opposed to float 32 stored on 4 bytes.
+The float precision is much lower but the training precision
+does not suffer too much.
+
+`FP8 Formats for Deep Learning <https://arxiv.org/abs/2209.05433>`_
+from NVIDIA, Intel and ARM introduces two types following
+`IEEE specifciations <https://en.wikipedia.org/wiki/IEEE_754>`_.
+First one is E4M3, 1 bit for the sign, 4 bits for the exponents and 3
+bits for the mantissa. Second one is E5M2, 1 bit for the sign,
+3 bits for the exponents and 2 for the mantissa. The first types
+is mostly used for the weights, the second one for the gradient.
+
+Second paper `8-bit Numerical Formats For Deep Neural Networks
+<https://arxiv.org/pdf/2206.02915.pdf>`_ introduces
+similar types. IEEE standard gives the same value
+to `+0` (or integer 0) and `-0` (or integer 128).
+They chose to give distinct float values to these two
+numbers. The paper experiments different split between
+exponent and mantissa and shows and E4M3 and E5M2 are
+the best ones.
+
+As a result, four new types were introduced in `onnx==1.15.0`
+to support a limited set of operators to enable computation
+with float 8.
+
+- `E4M3FN`: 1 bit for the sign, 4 bits for the exponents, 3 bits for the mantissa,
+  only nan values and no infinite values (FN),
+- `E4M3FNUZ`: 1 bit for the sign, 4 bits for the exponents, 3 bits for the mantissa,
+  only nan values and no infinite values (FN), no negative zero (UZ)
+- `E5M2`: 1 bit for the sign, 5 bits for the exponents, 2 bits for the mantissa,
+- `E5M2FNUZ`: 1 bit for the sign, 5 bits for the exponents, 2 bits for the mantissa,
+  only nan values and no infinite values (FN), no negative zero (UZ)
+
+The implementation is usually hardware dependant.
+NVIDIA implements `E4M3FN` and `E5M2` is its latest graphical processor.
+GraphCore does the same only with `E4M3FNUZ` and `E5M2FNUZ`.
+
+FP8 from IEEE
+=============
+
+Definition
+++++++++++
+
+:math:`S` stands for the sign. :math:`10_2` describe a number base 2.
+
+.. list-table:: Float8 types
+   :widths: 10 10 10
+   :header-rows: 1
+
+   * - 
+     - E4M3FN
+     - E5M2
+   * - Exponent bias
+     - 7
+     - 15
+   * - Infinities
+     -
+     - :math:`S.11111.00_2`
+   * - NaN
+     - :math:`S.1111.111_2`
+     - :math:`S.11111.\{01, 10, 11\}_2`
+   * - Zeros
+     - :math:`S.0000.000_2`
+     - :math:`S.00000.00_2`
+   * - Max
+     - :math:`S.1111.110_2`
+     - :math:`1.75 \times 2^{15}= 57344`
+   * - Min
+     - :math:`S.0000.001_2 = 2^{-9}`
+     - :math:`S.00000.01_2 = 2^{-16}`
+
+
+Let's denote the bit representation as :math:`S.b_6 b_5 b_4 b_3 b_2 b_1 b_0`.
+The float value is defined by the following expressions:
+
+.. list-table:: Float8 types values
+   :widths: 10 10 10
+   :header-rows: 1
+
+   * - 
+     - E4M3FN
+     - E5M2
+   * - exponent :math:`\neq` 0
+     - :math:`(-1)^S 2^{\sum_{i=3}^6 b_i 2^{i-3} - 7} \left( 1 + \sum_{i=0}^2 b_i 2^{i-3} \right)`
```

Comment:
I don't know if it's just me, but these Latex equations don't seem to be rendered properly in the generated file.

Reply:
The syntax is RST, it will properly render here: https://onnx.ai/onnx/index.html.

Reply:
Unfortanatly github lacks support for RST :math:.

For markdown files math latex syntax with $ will create render the latex. I think this is also works for RST files in github. 

example $S.b_6 b_5 b_4 b_3 b_2 b_1 b_0$